### ISSUE TITLE
go-envvar should recurse on nested structs

### ISSUE THREAD

**Issue by ntindall** [plaid/go-envvar#10]
`envvar.Parse()` does not support the following situation:

```go
type nestedEnvironment struct {
 testingValue `envvar:"TESTING_VALUE"`
}

type env struct {
  testingKey `envvar:"TESTING_KEY"`
  nestedEnvironment
}
```

envvar cannot support nested environment struct definitions. I think this could be supported by recursing on nested structs and testing if they had `envvar` struct tags, populating the nested struct appropriately.

**Comment by albrow** [comment]
Although it might be tricky to get right, I would suggest mimicking how Go handles nested structs (which the documentation refers to as "anonymous structs") in the [encoding/json package](https://golang.org/pkg/encoding/json/#Marshal). I expect this would lead to the least surprises.

**Comment by jeeyoungk** [comment]
Oh I wanted to implement this for a very long time.

**Comment by jeeyoungk** [comment]
For this, I've been thinking that nesting can support some kind of prefixing (Maybe this is going to be confusing)? I've been thinking something like:

```go
type nestedEnvironment struct {
 testingValue `envvar:"TESTING_VALUE"`
}

type env struct {
  testingKey `envvar:"TESTING_KEY"`
  nestedEnvironment `envvar_prefix:"NESTED_"`
}
```

In this case, `nestedEnvironment.testingValue` will be set via the envvar `NESTED_TESTING_VALUE`.

WDYT?

**Comment by jeeyoungk** [comment]
https://github.com/plaid/go-envvar/pull/13 i have a sample out.

**Comment by ntindall** [comment]
Fixed by #13 

published with [v1.0.0](https://github.com/plaid/go-envvar/releases/tag/v1.0.0)
